### PR TITLE
Robotino3.proto is modified to increase movement speed and accuracy and change orientation to FLU.

### PR DESCRIPTION
**Pull request by Thomas-Feldmeier**
Robotino3 is modified to be able to move fast (was very slow). Changed robot's coordinate system to x=forward y=left z=up.

proto/Robotino3.proto:
* changed robots coordinate system to x=front, y=left, z=up
* corrected position of wheels
* robot > Physics > centerOfMass 0 0 -0.1

proto/Robotino3Wheel.proto:
* device > RotationalMotor > maxVelocity 137.46 (allows max. speed 10 m/s forwards)
* boundingObject > Cylinder > height 0.01  (from 0.06, better robot rotation)

controller/robotino3.c, base.c, base.h:
* increased max speed from 0.1 m/s to 8 m/s
* increased demo movement speed from 0.05 m/s to 2 m/s
* corrected the movement calculation
* added acceleration

worlds/robotino3.wbt:
    WorldInfo > ContactProperties >
      material1 "WheelMat"
      coulombFriction 0 2 0
      bounce 0

(frictionRotation and forceDependentSlip have been removed)

https://cyberbotics.com/doc/guide/robotino3 should be changed to this new ContactProperties too.

ONLY USE IN WORLDS WITH WorldInfo.coordinateSystem "NUE"
(maybe the physics engine has an error in asymmetric friction with "ENU")

### CODE REVIEW

**Comment by omichel** [comment]
Thank you for this nice contribution!
Before we can merge it, we need to fix a few details.
I already fixed some clang format issue, but other issues are remaining.
For example, the CI is complaining that it misses an icon for the new wheel proto.
Can you add one?
You may want to use the [icon studio](https://github.com/cyberbotics/webots/tree/master/scripts/icon_studio) tool for this.

File: projects/robots/festo/robotino3/controllers/FastRobotino/FastRobotino.cpp
Comment (omichel) [inline projects/robots/festo/robotino3/controllers/FastRobotino/FastRobotino.cpp:34]: It's generally better to use explicit symbol names (variables, functions) rather than having to write comments to explain them.

```suggestion
static void moveForward(double vx, double vy, double angularVelocity) {
  const double wheelRadius = 0.063;
  vx /= wheelRadius;
  vy /= wheelRadius;
  angularVelocity *= 0.1826 / wheelRadius;  // distance from wheel center to robot center is 0.1826 m.
  motor[0]->setVelocity(vy + angularVelocity);
  motor[1]->setVelocity(-0.5 * vy - sqrt(0.75) * vx + angularVelocity);
  motor[2]->setVelocity(-0.5 * vy + sqrt(0.75) * vx + angularVelocity);
}
```

File: projects/robots/festo/robotino3/controllers/FastRobotino/FastRobotino.cpp
Comment (omichel) [inline projects/robots/festo/robotino3/controllers/FastRobotino/FastRobotino.cpp:38]:
```suggestion
  const double timeStep = robot->getBasicTimeStep();
```

File: projects/robots/festo/robotino3/controllers/FastRobotino/FastRobotino.cpp
Comment (omichel) [inline projects/robots/festo/robotino3/controllers/FastRobotino/FastRobotino.cpp:50]:
```suggestion
  const double maxAcceleration[3] = {10, 6, 20};
```

File: projects/robots/festo/robotino3/controllers/FastRobotino/FastRobotino.cpp
Comment (omichel) [inline projects/robots/festo/robotino3/controllers/FastRobotino/FastRobotino.cpp:82]:
```suggestion
    moveForward(actualSpeed[0], actualSpeed[1], actualSpeed[2]);
```

File: projects/robots/festo/robotino3/protos/Robotino3.proto
Comment (omichel) [inline projects/robots/festo/robotino3/protos/Robotino3.proto:12]: That seems wrong.

```suggestion
```

File: projects/robots/festo/robotino3/protos/Robotino3Wheel.proto
Comment (omichel) [inline projects/robots/festo/robotino3/protos/Robotino3Wheel.proto:12]: We are trying to avoid any abbreviations to make things self-explanatory.

```suggestion
  field SFFloat number 0
```

File: projects/robots/festo/robotino3/protos/Robotino3Wheel.proto
Comment (omichel) [inline projects/robots/festo/robotino3/protos/Robotino3Wheel.proto:6]: That looks wrong. This proto is about the Robotino3 wheel instead.

File: projects/robots/festo/robotino3/protos/Robotino3Wheel.proto
Comment (omichel) [inline projects/robots/festo/robotino3/protos/Robotino3Wheel.proto:58]: This DEF name is not used and seems to be useless.

```suggestion
              geometry IndexedFaceSet {
```

File: projects/robots/festo/robotino3/protos/Robotino3Wheel.proto
Comment (omichel) [inline projects/robots/festo/robotino3/protos/Robotino3Wheel.proto:21]: There is some indentation missing here.

**Review by omichel — CHANGES_REQUESTED** [review]
Here are a few suggestions for improvement.
Additionally, the C++ controller should be renamed to a snake case, e.g., `fast_robotino` instead of `FastRobotino`, to be consistent with other C++ controllers distributed with Webots.

File: projects/robots/festo/robotino3/protos/Robotino3Wheel.proto
Comment (Thomas-Feldmeier) [reply projects/robots/festo/robotino3/protos/Robotino3Wheel.proto:21]: Hi Olivier Michel, thanks for your help!

I changed the Robotino3Wheel.proto to # tags: hidden so it theory it should not need any icon file anymore because it is not visible to users. (like kuka/youbot/protos/Arm0Mesh.proto needs no icons file)

I added the FastRobotino.cpp controller code to the original robotino3.c, base.c and base.h controller code.

I changed in robotino3.wbt the ContactProperties values.

On https://cyberbotics.com/doc/guide/robotino3 the ContactProperties should be changed too:

   coulombFriction 0.0,  2.0,  0.0

  <remove frictionRotation>

  <remove forceDependentSlip>

I will soon upload these changes. (except the https page changes)

Comment (omichel) [reply projects/robots/festo/robotino3/protos/Robotino3Wheel.proto:20]: > I changed the Robotino3Wheel.proto to # tags: hidden so it theory it should not need any icon file anymore because it is not visible to users. (like kuka/youbot/protos/Arm0Mesh.proto needs no icons file)

That's right, sorry I overlooked it.

Comment (omichel) [reply projects/robots/festo/robotino3/protos/Robotino3Wheel.proto:20]: > On https://cyberbotics.com/doc/guide/robotino3 the ContactProperties should be changed too:

You should do it in this PR as well, simply modify [this file](https://github.com/cyberbotics/webots/blob/master/docs/guide/robotino3.md). Once merged into the master branch, it will be automatically published on the web site.

Comment (omichel) [reply projects/robots/festo/robotino3/protos/Robotino3Wheel.proto:20]: To save you from clang format errors, you should probably install a clang format linter in your source code editor as explained [here](https://github.com/cyberbotics/webots/wiki/CPP-Coding-Style#tools). With it, your C++ code will be automatically formatted to pass the clang format CI tests.

File: projects/robots/festo/robotino3/controllers/robotino3/base.c
Comment (omichel) [inline projects/robots/festo/robotino3/controllers/robotino3/base.c:84]:
```suggestion
  const double timeStep = wb_robot_get_basic_time_step();
```

File: projects/robots/festo/robotino3/controllers/robotino3/robotino3.c
Comment (omichel) [inline projects/robots/festo/robotino3/controllers/robotino3/robotino3.c:42]:
```suggestion
static double time_step;
```

File: projects/robots/festo/robotino3/controllers/robotino3/robotino3.c
Comment (omichel) [inline projects/robots/festo/robotino3/controllers/robotino3/robotino3.c:74]:
```suggestion
  if (wb_robot_step(time_step) == -1) {
```

File: projects/robots/festo/robotino3/controllers/robotino3/robotino3.c
Comment (omichel) [inline projects/robots/festo/robotino3/controllers/robotino3/robotino3.c:235]:
```suggestion
  time_step = wb_robot_get_basic_time_step();
```

File: projects/robots/festo/robotino3/controllers/robotino3/robotino3.c
Comment (omichel) [inline projects/robots/festo/robotino3/controllers/robotino3/robotino3.c:240]:
```suggestion
    wb_camera_enable(camera, time_step);
```

File: projects/robots/festo/robotino3/controllers/robotino3/robotino3.c
Comment (omichel) [inline projects/robots/festo/robotino3/controllers/robotino3/robotino3.c:245]:
```suggestion
    wb_distance_sensor_enable(infrared_sensors[i], time_step);
```

File: projects/robots/festo/robotino3/controllers/robotino3/robotino3.c
Comment (omichel) [inline projects/robots/festo/robotino3/controllers/robotino3/robotino3.c:263]:
```suggestion
  wb_keyboard_enable(time_step);
```

**Review by omichel — CHANGES_REQUESTED** [review]
Almost there...

File: projects/robots/festo/robotino3/protos/Robotino3.proto
Comment (omichel) [inline projects/robots/festo/robotino3/protos/Robotino3.proto:34]: This is too much information to put in the PROTO file, we can get rid of the change log here, as history can be found on your PR. I would remove this text, and just keep your name as a contributor. The warning about "ENU" should go the documentation page.

**Review by omichel — APPROVED** [review]
Thank you for this very nice contribution!

**Comment by Thomas-Feldmeier** [comment]
Can you please merge this PR so i can test it tomorrow in the nightly build?

**Comment by omichel** [comment]
Now that you accepted the invitation to the [committers group](https://github.com/orgs/cyberbotics/teams/committers/members), you should be able to merge it yourself, isn't it? Otherwise I will merge it.

**Pull request merged**
